Parse/PFQuery.m

```diff
@@ -843,7 +843,7 @@ - (BFTask *)_findObjectsAsyncForQueryState:(PFQueryState *)queryState after:(BFT
     }] continueWithBlock:^id(BFTask *task) {
         @strongify(self);
```

Comment:
Move the `nil` check here and bail out early, if it's nil, in order not to reference `self` inside `@synchronized()`